restapi/README-docsapi.md

```diff
@@ -0,0 +1,168 @@
+# Stargate Documents API
+
+The Documents API is built into the REST API module, an OSGi bundle that exposes various CRUD services over HTTP.
+Specifically, the Documents API allows for storage of documents in the JSON format using C* tables (called "collections" in this context).
+
+Here is a brief overview of the service (for more details, refer to the [Stargate
+online docs]).
+
+* There are endpoints for writing, updating, and deleting documents in [DocumentResource]
+* For reading documents and searching through documents and collections, there are endpoints in [ReactiveDocumentResource]
+* CRUD operations for collections exist in [CollectionsResource], and operations for modifying keyspace information (called "namespaces" in this context) are in [NamespacesResource]
+* Operations for attaching a JSON Schema (see [Json schema info] for more details) are in [JsonSchemaResource]
+
+## Entry points
+
+[RestApiActivator] manages the starting and stopping of the OSGi bundle, which runs both the Documents API and the more general REST API.
+
+[Server] is the HTTP server that exposes the Documents API services as REST resources. It is implemented with
+[Dropwizard](https://www.dropwizard.io/en/latest/).
+
+### Authentication
+
+Every HTTP request performs token-based authentication. This is done by calling [Db]#getDocDataStoreForToken explicitly.
+This method will either return an instance of [DocumentDB] which can be used for querying Cassandra and contains the authenticated subject,
```

Comment:
```suggestion
This method will either return an instance of [DocumentDB] which can be used for querying Cassandra and contains the authenticated subject,
```
```suggestion
This method will either return an instance of [DocumentDB] which can be used for querying the underlying persistence and contains the authenticated subject,
```

Nit: Since it's C* and DSE.